network/dag/signing.go

```diff
@@ -66,6 +72,13 @@ func (d documentSigner) Sign(input UnsignedDocument, signingTime time.Time) (Doc
 		if err != nil {
 			return nil, fmt.Errorf(errSigningDocumentFmt, err)
 		}
+		// The JWX library is fine with creating a JWK for a private key (including the private exponents), so
+		// we want to make sure the resolver didn't accidentally return a private key which just conforms to the
+		// crypto.PublicKey interface. Since that would lead to the node leaking its private key material in the
+		// resulting JWS which would be very, very bad.
+		if asSigner, ok := keyAsPublicKey.(crypto2.Signer); ok {
```

Comment:
Can't we do this at another place? Getting a pub key and checking if it's a pub key feels off

Reply:
Moved it as a check to `crypto.signJWS`, it just refuses to sign when the `jwk` field contains a private key.